it_zwierzak/adoption_announcement/views.py

```diff
@@ -1,22 +1,45 @@
 from django.shortcuts import render
-from django.http import HttpResponseRedirect, HttpResponse
+from django.views.generic import FormView
+from .forms import AdoptionForm
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
 
-from .templates import forms
 
-def index(request):
-    return HttpResponse("Ogłoszenie Adopcji")
-
-def base(request):
-    return render(request, 'adoption_announcement/base.html')
 
+# zaadoptowane
 def adopted(request):
-    return render(request, 'adoption_announcement/adopted.html')
+    return render(request, 'adoption_announcement/adopted_animals.html')
+
 
+# przeniesienie do strony z kafelkami adopcji
 def adoptions(request):
-    return render(request, 'adoption_announcement/adoptions.html')
+    return render(request, 'adoption_announcement/animals_to_adoption.html')
+
 
+# przeniesienie do konkretnego ogłoszenia zwierzęcia, po id z bazy danych
 def adoption_card(request):
-    return render(request, 'adoption_announcement/adoption_card.html')
+    return render(request, 'adoption_announcement/announcement_detail.html')
+
 
 def adoption_form(request):
-    return render(request, 'adoption_announcement/adoption_form.html')
+    return render(request, 'adoption_announcement/adoption_application.html')
+
+
+class AdoptionApplicationView(LoginRequiredMixin, FormView): #tylko dla zalogowanych
+    template_name = 'adoption_announcement/adoption_application.html'
+    form_class = AdoptionForm
+    success_url = '/animals_to_adoption'
+
+def adoption_application_view(request):
+    form = AdoptionForm(request.POST or None)
+    if form.is_valid():
+        form.save()
+
+    context = {
+        'form': form
+    }
+    return render(request, 'adoption_announcement/adoption_application.html', context)
+
+@login_required
+def private_page(request):
```

Comment:
dajmy lepszą nazwę jeśli to coś ma zostać, a jeśli było tylko do testów to do usunięcia

Reply:
zmiana na adopted_application_for_logged_in